src/lsdb/loaders/hipscat/read_hipscat.py

```diff
@@ -66,6 +73,79 @@ def read_hipscat(
     return loader.load_catalog()
 
 
+def read_hipscat_subset(
+    path: str,
+    catalog_type: Type[Dataset] | None = None,
+    storage_options: dict | None = None,
+    search_filter: AbstractSearch | None = None,
+    order: int | None = None,
+    n_pixels: int | None = None,
+    **kwargs,
+) -> Dataset:
+    """Load a catalog subset from a HiPSCat formatted catalog.
+
+    To be used with a filter or (order, n_files), but not both.
+
+    Typical usage example, where we load a catalog from a cone search:
+        lsdb.read_hipscat_subset(
+            path="./my_catalog_dir",
+            catalog_type=lsdb.Catalog,
+            columns=["ra","dec"],
+            filter=lsdb.core.search.ConeSearch(ra, dec, radius_arcsec),
+        )
+
+    Typical usage example, where we load a catalog from a pre-defined number of files:
+         lsdb.read_hipscat_subset(
+            path="./my_catalog_dir",
+            catalog_type=lsdb.Catalog,
+            columns=["ra","dec"],
+            order=5,
+            n_pixels=10,
+        )
+
+    Args:
+        path (str): The path that locates the root of the HiPSCat catalog
+        catalog_type (Type[Dataset]): Default `None`. By default, the type of the catalog is loaded
+            from the catalog info and the corresponding object type is returned. Python's type hints
+            cannot allow a return type specified by a loaded value, so to use the correct return
+            type for type checking, the type of the catalog can be specified here. Use by specifying
+            the lsdb class for that catalog.
+        storage_options (dict): Dictionary that contains abstract filesystem credentials
+        search_filter (Type[AbstractSearch]): The filter method to be applied to the catalog.
+            By default, all files are considered.
+        n_pixels (int): The number of files to read, in case no filter was specified.
+            By default, all files are considered.
+        order (int): The order of the files to read, in case n_files was specified.
+            By default, the largest catalog order is used.
```

Comment:
Using order like this feels unintuitive to me. I'd expect that by default setting `n_pixels` would get the first n pixels, not the first n pixels of the highest order.